### ISSUE TITLE
Image only blurs when run through INetwork

### ISSUE THREAD

**Issue by cversteeg** [titu1994/Neural-Style-Transfer#12]
I am trying to run the INetwork keras code, but am running into an issue where the image only seems to blur, but does not take any style from the other image. I have tried tweaking the parameters as suggested in your guide to no seeming effect. 

The images that I am using are 
![haydog](https://cloud.githubusercontent.com/assets/15896921/19140662/a1491c0e-8b54-11e6-84ac-33aae77e723d.JPG)

for the source and

![haystacks-in-provence](https://cloud.githubusercontent.com/assets/15896921/19140669/ab5b2c82-8b54-11e6-9432-7a0f06cf33b7.jpg)

for the style.

The most recent parameters attempted are 

content weight: .025
style_weight: 1.0
style_scale: 1
tv_weight: .000085
content layer conv5_2
max pooling

The image looks like this after 1
![haydoggie_at_iteration_10](https://cloud.githubusercontent.com/assets/15896921/19140765/67e1e3aa-8b55-11e6-8743-96f74b49a598.png)
0 iterations.

**Comment by titu1994** [comment]
@cversteeg I'll try running the code on my pc and see if I can reproduce your problem.

**Comment by titu1994** [comment]
@cversteeg Just finished 10 iterations on it. The results aren't great but with additional parameter tuning I an quite sure it should give good results. 

![test_at_iteration_10](https://cloud.githubusercontent.com/assets/3048602/19140973/37eebbc8-8baf-11e6-9554-2c8cce8de63a.png)

Can I ask if you are using the latest version of the script? Also, I have seen blurring in cases where content weight is too strong in comparison to style weight but this isn't the case here.

**Comment by cversteeg** [comment]
I pulled and ran it again with the default parameters, but no improvement. 
This is the output on my end. There is one divide by zero error in there, but I am not sure if it's significant.

Using TensorFlow backend.
Model loaded.
Start of iteration 1
/home/chris/Neural-Style-Transfer/INetwork.py:375: RuntimeWarning: invalid value encountered in double_scalars
  improvement = (prev_min_val - min_val) / prev_min_val \* 100
Current loss value: -1.01924e+08  Improvement : nan %
Rescaling Image to (400, 533)
Image saved as /home/chris/Pictures/learned_at_iteration_1.png
Iteration 1 completed in 12s
Start of iteration 2
Current loss value: 8105.98  Improvement : 100.008 %
Rescaling Image to (400, 533)
Image saved as /home/chris/Pictures/learned_at_iteration_2.png

Is is possible that the weights aren't loading properly from the vgg16 model?

**Comment by titu1994** [comment]
@cversteeg I've been meaning to get that error checked. It has no effect on output quality though. It's simply a calculation error in the improvement score.

As to your loss value, that is absurdly low. Even at the 10th epoch, my loss value was near the 14~ million range. Your's however gets to 8100 in 2 epochs.

I'm on Windows and therefore am not able to test Tensorflow, but the script should automatically download and use the Tensorflow weights from the Keras repository of weights. It would make sense however, since I just tested this on Theano.

**Comment by cversteeg** [comment]
I'll try to change my backend to see if that helps.

Thanks for the help!
Chirs

**Comment by titu1994** [comment]
@cversteeg  Can you print your .keras/keras.json config? 

I think I found the error, in the fact that, if you use image_dim_ordering = "th" but with backend = "tensorflow", then I need to perform a correction to the convolution kernels. 

Edit:
In anycase, I updated the scripts now, so they will warn users about using the wrong image_dim_ordering with the Tensorflow backend.

**Comment by cversteeg** [comment]
This was my keras.json when I encountered the issue

{
    "floatx": "float32",
    "backend": "tensorflow",
    "image_dim_ordering": "tf",
    "epsilon": 1e-07
}

I changed my backend to theano and it is working now, though I have to setup my GPU to play nicely with Theano.

I appreciate all of the help!

![learned_at_iteration_2](https://cloud.githubusercontent.com/assets/15896921/19141862/98abb070-8b5e-11e6-931e-5e00a58621d0.png)

**Comment by titu1994** [comment]
@cversteeg No problem. But even your keras.json is correct. 

I have no idea then why this doesn't work properly with Tensorflow backend. The script is originally from the keras example, and I only exposed the weight parameters. If that script works on tensorflow, it should stand to reason that this should too. 

Anyway, glad it works on theano at least.
